src/components/PostsList.tsx

```diff
@@ -2,5 +1,22 @@
+import { Post } from '../types/Post';
 
-export const PostsList: React.FC = () => (
+type Props = {
+  posts: Post[];
+  setPost: (post: Post) => void;
+  openPostDetails: number;
+  setOpenPostDetails: (openPostDetails: number) => void;
+  getCommentsPost: (post: Post) => void;
+  setWriteComment: (writeComment: boolean) => void;
+};
+
+export const PostsList: React.FC<Props> = ({
+  posts,
+  setPost,
+  openPostDetails,
+  setOpenPostDetails,
+  getCommentsPost,
+  setWriteComment,
+}) => (
```

Comment:
Ensure that all functions and props have explicit type definitions to enhance type safety and prevent potential runtime errors.